client/src/main/java/org/devfleet/esi/impl/service/ESICorporationServiceImpl.java

```diff
@@ -10,68 +14,73 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
-class CorporationAPIImpl {
+import static java.util.Collections.emptyList;
 
-    private static Logger LOG = LoggerFactory.getLogger(CorporationAPIImpl.class);
+class ESICorporationServiceImpl implements ESICorporationService {
 
-    private final LiveApi liveApi;
+    private static Logger LOG = LoggerFactory.getLogger(ESICorporationServiceImpl.class);
+
+    private final CorporationApi api;
 
     private final String datasource;
 
-    public CorporationAPIImpl(
+    public ESICorporationServiceImpl(
             String datasource,
-            LiveApi liveApi) {
-        this.liveApi = liveApi;
+            CorporationApi api) {
+        this.api = api;
         this.datasource = datasource;
     }
 
-    public Corporation getCorporation(Long corpID) {
+    @Override
+    public Optional<CorporationData> tryFetchCorporation(CorporationKey key) {
         try {
-            Corporation corporation =
-                    EsiTransformer.transform(
-                        corpID,
-                        this.liveApi
-                        .getCorporationsCorporationId(corpID.intValue(), this.datasource)
-                        .execute()
-                        .body());
+            CorporationData corporation =
+                    ESITransformer.transform(
+                        key,
+                        api
+                                .getCorporationsCorporationId(key.getId(), datasource)
+                                .execute()
+                                .body());
             addPortraits(corporation);
             addHistory(corporation);
-            return corporation;
+            return Optional.of(corporation);
         }
         catch (IOException e) {
             LOG.debug(e.getLocalizedMessage(), e);
             LOG.warn(e.getLocalizedMessage());
-            return null;
+            return Optional.empty();
         }
     }
 
-    public List<Corporation.Member> getMembers(Long corpID) {
+    @Override
+    public List<CorporationMember> getMembers(CorporationKey key) {
         try {
             List<GetCorporationsCorporationIdMembers200Ok> members =
-                this.liveApi.getCorporationsCorporationIdMembers(corpID.intValue(), this.datasource)
-                    .execute()
-                    .body();
-            final List<Corporation.Member> r = new ArrayList<>(members.size());
+                api
+                        .getCorporationsCorporationIdMembers(key.getId(), datasource)
+                        .execute()
+                        .body();
+            final List<CorporationMember> r = new ArrayList<>(members.size());
             for (GetCorporationsCorporationIdMembers200Ok m: members) {
-                r.add(EsiTransformer.transform(m));
+                r.add(ESITransformer.transform(m));
```

Comment:
I can't believe there is someone out there who is more adamant on proper usage of capitalization in abbreviations.

+1 for ESI :)